zips/zip-0233.md

```diff
@@ -19,224 +20,146 @@ The key words "MUST", "SHOULD", "SHOULD NOT", "MAY", "RECOMMENDED", "OPTIONAL",
-can be done by:
-- ZEC fund holders, in non-coinbase transactions;
-- Zcash miners, in coinbase transactions.
+**Note:** Older transaction versions can continue to be supported after a
+network upgrade, but burning is not possible for these transactions. For
+example, NU5 supports both v4 and v5 transaction formats, for both coinbase and
+non-coinbase transactions.
 
-In transaction versions without this new field:
-- unclaimed miner fees and rewards in **coinbase transactions** are re-defined
-  as deposits into the sustainability fund, and
-- there are no sustainability fund deposits from non-coinbase transactions.
+## Consensus Rules
 
-Note: older transaction versions can continue to be supported after a network
-upgrade. For example, NU5 supports both v4 and v5 transaction formats, for both
-coinbase and non-coinbase transactions.
+The burned value must now be considered when calculating the total output value
+of a transaction. It should be treated similarly to a transparent output,
+except that there is no way for this value to be used as an input in a future
+transaction.
 
-### `zsfDeposit` fields in transactions
+## Digests
 
-Each transaction can dedicate some of its excess funds to the ZSF, and the
-remainder becomes the miner fee, with any excess miner fee/reward going to the
-ZSF
+The transaction digest algorithm defined in ZIP 244 [4] is to be modified for v7
+transactions as follows:
 
-This is achieved by adding a new field to the common transaction fields
-[#zip-0225-transaction-format]:
+A section is inserted, specified to read:
 
-- `zsfDeposit : u64 [zatoshi]`
+> Section T.5: `burn_amount_digest`
+>
+> A BLAKE2b-256 hash of the following values:
+>
+> ```
+> T.5a: burnAmount (8-byte little-endian value)
+> ```
+>
+> The personalization field of this hash is set to:
+>
+> ```
+> ZTxBurnAmnt_Hash
+> ```
 
-The `ZSF_BALANCE[H]` for a block at height `H` can be calculated given a value
-of `ZSF_BALANCE[H-1]` and the set of transactions contained in that block.
-First, the `ZSF_DEPOSIT[H]` is calculated based solely on `ZSF_BALANCE[H-1]`.
-This is subtracted from the previous block's balance to be distributed as part
-of the block reward. Second, the sum of all the `ZSF_DEPOSIT` fields of all
-transactions in the block is added to the balance.
 
+`txid_digest` is to be modified to append T.5 (`burn_amount_digest`) to the list
+of values.
 
-### Non-Coinbase Transactions
+The signature digest algorithm defined in ZIP 244 [4] is to be modified for v7
+transactions as follows:
 
-If the `ZSF_DEPOSIT` field is not present in an older transaction version, it
-is defined to be zero for non-coinbase transactions.
+A section is inserted, specified to read:
 
-#### Consensus Rule Changes
+> Section S.5: `burn_amount_digest`
+>
+> A BLAKE2b-256 hash of the following values:
+>
+> ```
+> S.5a: burnAmount (8-byte little-endian value)
+> ```
+>
+> The personalization field of this hash is set to:
+>
+> ```
+> ZTxBurnAmnt_Hash
+> ```
```

Comment:
Again, this will also be part of the header digest (unless there is a specific rationale for a separate section.)

Reply:
Will do :+1: